sdks/python/apache_beam/transforms/stats.py

```diff
@@ -576,17 +633,34 @@ def _interpolate(self, i_buffers, count, step, offset):
     weighted_element = next(sorted_elem)
     current = weighted_element[1]
     j = 0
-    while j < count:
-      target = j * step + offset
-      j = j + 1
-      try:
-        while current <= target:
-          weighted_element = next(sorted_elem)
-          current = current + weighted_element[1]
-      except StopIteration:
-        pass
-      new_elements.append(weighted_element[0])
-    return new_elements
+    if self._weighted:
```

Comment:
I wonder if it's possible to do this without duplicating the code in both branches? I understand that separating branches may help performance, but it decreases on readability...  WDYT?

Reply:
I tried to avoid affecting the performance of the existing codepath, but after a closer look I think the overhead caused by the if in the loop is negligible compared to the time taken by the inner loop and append. Merged the branches.